catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -0,0 +1,445 @@
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+import org.apache.shiro.SecurityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.io.ByteSource;
+import com.google.common.io.Files;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.content.StorageException;
+import ddf.catalog.content.StorageProvider;
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.data.impl.ContentItemImpl;
+import ddf.catalog.content.operation.CreateStorageRequest;
+import ddf.catalog.content.operation.CreateStorageResponse;
+import ddf.catalog.content.operation.UpdateStorageRequest;
+import ddf.catalog.content.operation.UpdateStorageResponse;
+import ddf.catalog.content.operation.impl.CreateStorageRequestImpl;
+import ddf.catalog.core.versioning.MetacardVersion;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.Operation;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.CreateResponseImpl;
+import ddf.catalog.source.CatalogProvider;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+
+public class Historian {
+    public static final String ALREADY_VERSIONED = "already-versioned";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Historian.class);
+
+    private static final String HISTORY_METACARDS = "history-metacards";
+
+    private boolean historyEnabled = true;
+
+    private List<StorageProvider> storageProviders;
+
+    private List<CatalogProvider> catalogProviders;
+
+    private CatalogFramework catalogFramework;
+
+    public Historian() {
+    }
+
+    private final Map<String, List<Callable<Boolean>>> staged = new HashMap<>();
+
+    public CreateRequest versionCreate(CreateRequest createRequest) throws IngestException {
+        if (doSkip(createRequest)) {
+            return createRequest;
+        }
+
+        List<Metacard> updatedMetacardList = createRequest.getMetacards()
+                .stream()
+                .filter(MetacardVersion::isNotVersion)
+                .map(m -> new MetacardVersion(m,
+                        MetacardVersion.Action.CREATED,
+                        SecurityUtils.getSubject()))
+                .collect(Collectors.toList());
+
+        if (!updatedMetacardList.isEmpty()) {
+            catalogProvider().create(new CreateRequestImpl(updatedMetacardList));
+        }
+
+        return createRequest;
+    }
+
+    public UpdateResponse versionUpdate(UpdateResponse updateResponse)
```

Comment:
Because in the update case we could just simply wait till the update was successfull and pull the metacards to version from the update response.
done